README.md

```diff
@@ -203,21 +203,29 @@ The following [AWS CloudFormation Intrinsic Functions](http://docs.aws.amazon.co
 - [ ] Fn::GetAZs   
 - [ ] Fn::ImportValue
 
-Any unsupported intrinsic functions will return `nil`.
+Any unsupported intrinsic functions will return `nil`. 
 
-#### Resolving References (Ref)
+**NOTE**: Properties that resolve to `nil` will be omitted from the output. For example:
 
-The intrinsic 'Ref' function as implemented will resolve all of the [pseudo parameters](http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference-ref.html) such as `AWS::AccountId` with their default value as listed on [the bottom of this page](http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference-ref.html).
+```
+Parameters:
+  RuntimeParameter:
+    Type: String
+
+MyFunction:
+  Type: AWS::Serverless::Functin
+  Properties:
+    Runtime: !Ref RuntimeParameter
+```
 
-If a reference is not a pseudo parameter, GoFormation will try to resolve it within the AWS CloudFormation template. **Currently, this implementation only searches for `Parameters` with a name that matches the ref, and returns the `Default` if it has one.**
+Since `RuntimeParameter` does not have a default value, the !Ref will be resolved to nil. When a property resolves to `nil`, it will be omitted from the struct returned by the unmarshal method.
```

Comment:
This isn't strictly true, the struct will still have the property, it will just = nil - which would be the expected behaviour. As such, i'm not sure we need to add this.